web/gridt/resources/user.py

```diff
@@ -45,4 +44,22 @@ def post(self):
         
         if current_identity.verify_password(res["old_password"]):
             current_identity.hash_password(res["new_password"])
-            return ({"message": "Successfully changed password."}, 200)
\ No newline at end of file
+            return ({"message": "Successfully changed password."}, 200)
+
+class ChangeEmailResource(Resource):
+    schema = ChangeEmailSchema()
+
+    @jwt_required()
+    def post(self):
+        try:
+            res = self.schema.load(request.get_json())
+        except ValidationError as error:
+            field = list(error.messages.keys())[0]
+            return ({"message": f"{field}: {error.messages[field][0]}"}, 400)
+
+        if not current_identity.verify_password(res["password"]):
+            return ({"message": "Failed to identify user with given password."}, 400)
```

Comment:
I think it could have been more elegant to leave this to the Schema.

Reply:
Not sure if I want that to be a pull-request stopper, but do want to comment.

Reply:
It would allow for the testing to be done entirely on the schema tests as well, which I think is appropriate.

Reply:
The schema does not accept flask's current_identity as a User object, causing any attempts at verifying the current user's password in the schema to fail.

Reply:
Can you expand on this a little bit? I don't think I fully understand.